src/main/java/com/uid2/client/PublisherUid2Client.java

```diff
@@ -62,16 +57,32 @@ public TokenGenerateResponse generateTokenResponse(TokenGenerateInput tokenGener
 
 
         try (Response response = client.newCall(request).execute()) {
-            if (!response.isSuccessful()) {
-                throw new Uid2Exception("Unexpected code " + response);
-            }
-
-            String responseString = response.body() != null ? response.body().string() : "";
+            final String responseString = getResponse(response);
             return publisherUid2Helper.createTokenGenerateResponse(responseString, envelope);
         } catch (IOException e) {
             throw new Uid2Exception("Error communicating with api endpoint", e);
         }
     }
+    
+    private static String getResponse(Response response) {
+
+        String responseString = "";
+
+        try {
+            if(response == null) {
+                responseString = "empty response";
```

Comment:
when would that "empty response" string be consumed?

Reply:
when `response ` is null which is never but just that to be defensive....

Reply:
Yeah I'm just wondering if we should be throwing instead

Reply:
ok changed to throw